HaxeManual/05-expressions.tex

```diff
@@ -169,11 +169,242 @@ \section{Constants}
+
+\paragraph{Numeric comparison operators}
+
+\begin{center}
+\begin{tabular}{| l | l | l | l | l |}
+	\hline
+	Operator & Operation & Operand 1 & Operand 2 & Result type \\ \hline
+	\expr{==} & equal & \type{Float/Int} & \type{Float/Int} & \type{Bool} \\
+	\expr{!=} & not equal & \type{Float/Int} & \type{Float/Int} & \type{Bool} \\
+	\expr{<} & less than & \type{Float/Int} & \type{Float/Int} & \type{Bool} \\
+	\expr{<=} & less than or equal & \type{Float/Int} & \type{Float/Int} & \type{Bool} \\
+	\expr{>} & greater than & \type{Float/Int} & \type{Float/Int} & \type{Bool} \\
+	\expr{>=} & greater than or equal & \type{Float/Int} & \type{Float/Int} & \type{Bool}
+\end{tabular}
+\end{center}
+
+\paragraph{String comparison operators}
+
+\begin{center}
+\begin{tabular}{| l | l | l | l | l |}
+	\hline
+	Operator & Operation & Operand 1 & Operand 2 & Result type \\ \hline
+	\expr{==} & equal & \type{String} & \type{String} & \type{Bool} \\
+	\expr{!=} & not equal & \type{String} & \type{String} & \type{Bool} \\
+	\expr{<} & lexicographically before & \type{String} & \type{String} & \type{Bool} \\
+	\expr{<=} & lexicographically before or equal & \type{String} & \type{String} & \type{Bool} \\
+	\expr{>} & lexicographically after & \type{String} & \type{String} & \type{Bool} \\
+	\expr{>=} & lexicographically after or equal & \type{String} & \type{String} & \type{Bool}
+\end{tabular}
+\end{center}
+
+\type{String} is an immutable type in Haxe, so two separate instances of the same string are considered equal.
+
+\paragraph{Equality operators}
+
+\begin{center}
+\begin{tabular}{| l | l | l | l | l |}
+	\hline
+	Operator & Operation & Operand 1 & Operand 2 & Result type \\ \hline
+	\expr{==} & equal & any & same as operand 1 & \type{Bool} \\
+	\expr{!=} & not equal & any & same as operand 1 & \type{Bool}
+\end{tabular}
+\end{center}
+
+\emph{For enums:}
+
+\begin{description}
+	\item[Enums without parameters] always represent the same value, so \expr{MyEnum.A == MyEnum.A}. 
+	\item[Enums with parameters] can be compared with \expr{a.equals(b)} (which is short for \expr{Type.enumEquals()}).
+\end{description}
+
+\emph{Dynamic:}
+
+Comparison involving at least one \type{Dynamic} value is unspecified and platform-specific.
+
+\paragraph{Miscellaneous operators}
+
+\begin{center}
+\begin{tabular}{| l | l | l | l | l |}
+	\hline
+	Operator & Operation & Operand 1 & Operand 2 & Result type \\ \hline
+	\expr{...} & interval (see \tref{range iteration}{expression-for}) & \type{Int} & \type{Int} & \type{IntIterator} \\
+	\expr{=>} & arrow (see \tref{map}{expression-map-declaration}, \tref{key-value iteration}{expression-for}, \tref{map comprehension}{lf-map-comprehension}) & any & any & -
+\end{tabular}
+\end{center}
+
+
+\subsection{Ternary Operator}
+\label{expression-operators-ternary}
+
+Haxe supports the ternary conditional operator \expr{cond ? whenTrue : whenFalse}:
```

Comment:
Since we're listing types everywhere else, it would be nice to mention the types for this as well.